x-pack/plugins/siem/public/components/url_state/index_mocked.test.tsx

```diff
@@ -0,0 +1,134 @@
+
+      const newUrlState = {
+        ...mockProps.urlState,
+        [CONSTANTS.timerange]: {
+          global: {
+            [CONSTANTS.timerange]: {
+              from: 0,
+              fromStr: 'now-24h',
+              kind: 'relative',
+              to: 1,
+              toStr: 'now',
+            },
+            linkTo: ['timeline'],
+          },
+          timeline: {
+            [CONSTANTS.timerange]: {
+              from: 0,
+              fromStr: 'now-24h',
+              kind: 'relative',
+              to: 1,
+              toStr: 'now',
+            },
+            linkTo: ['global'],
+          },
+        },
+      };
+
+      wrapper.setProps({ urlState: newUrlState });
+      wrapper.update();
+      expect(mockHistory.replace.mock.calls[2][0]).toStrictEqual({
+        hash: '',
+        pathname: '/network',
+        search:
+          '?_g=()&timerange=(global:(linkTo:!(timeline),timerange:(from:0,fromStr:now-24h,kind:relative,to:1,toStr:now)),timeline:(linkTo:!(global),timerange:(from:0,fromStr:now-24h,kind:relative,to:1,toStr:now)))',
+        state: '',
+      });
+    });
+    test('kql query redux state updates the url', () => {
+      mockProps = getMockPropsObj({
+        page: CONSTANTS.networkPage,
+        examplePath: '/network',
+        namespaceLower: 'network',
+        type: networkModel.NetworkType.page,
+      }).noSearch.definedQuery;
+      const wrapper = shallow(<UrlStateContainerLifecycle {...mockProps} />);
+
+      const newUrlState = {
+        [CONSTANTS.kqlQuery]: {
+          [CONSTANTS.networkPage]: {
+            filterQuery,
+            type: networkModel.NetworkType.page,
+            queryLocation: CONSTANTS.networkPage,
+          },
+        },
+      };
+
+      wrapper.setProps({ urlState: newUrlState });
+      wrapper.update();
+
+      expect(mockHistory.replace.mock.calls[0][0]).toStrictEqual({
+        hash: '',
+        pathname: '/network',
+        search:
+          "?_g=()&kqlQuery=(filterQuery:(expression:'host.name:%22siem-es%22',kind:kuery),queryLocation:network.page,type:page)",
+        state: '',
+      });
+    });
+  });
+
+  describe('handleInitialize', () => {
+    describe('Redux updates URL state', () => {
+      describe('kqlQuery and timerange url state is set when not defined on component mount', () => {
+        test.each(testCases)('%o', (page, namespaceLower, namespaceUpper, examplePath, type) => {
+          mockProps = getMockPropsObj({ page, examplePath, namespaceLower, type }).noSearch
+            .undefinedQuery;
+          shallow(<UrlStateContainerLifecycle {...mockProps} />);
+
+          expect(mockHistory.replace.mock.calls[0][0]).toEqual({
+            hash: '',
+            pathname: examplePath,
+            search: `?_g=()&kqlQuery=(filterQuery:!n,queryLocation:${page},type:${type})`,
+            state: '',
+          });
+
+          // @ts-ignore property mock does not exists
```

Comment:
I tried it and you do not seem to need to have the ts-ignore here

Reply:
yep. good catch!